.github/ISSUE_TEMPLATE/question.md

```diff
@@ -0,0 +1,37 @@
+---
+name: Code Question
+about: Ask a question about the code
+title: ''
+labels: 'Type: Question'
+assignees: ''
+
+---
+
+<!--
+Thank you for taking an interest in the OpenZFS codebase.
+
+Please be aware that most questions are preferably asked in the mailinglist first.
+This form is primarily meant for ask questions about the code itself.
+
+Please also check our issue tracker before opening a new question.
+Filling out the following template will help other contributors better understand your proposed feature.
+-->
+
+### Ask your question!
+
+<!--
+Provide a clear and concise question and please add as much information as possible.
+-->
+
+### Which portion of the codebase does your question involve?
+
+<!--
+Optional: Please describe what portion of the codebase your issue involved.
+Example: "Testsuite", "Buildbots", "CLI" or a codesnippet
```

Comment:
How about adding, ", etc" to the end to indicate this isn't an exhaustive list.

" Example: "Testsuite", "Buildbots", "CLI", code snippet, etc"

(I believe codesnippet is two words)

Reply:
Indeed!